### ISSUE TITLE
Discrete VSCode configs by technology stack

### ISSUE THREAD

**Issue by RodneyBollinger** [Microsoft/vscode#67181]
I would like to see the ability to configure multiple launch configurations of VSCode that only include the extensions I need for a given stack (e.g. Node/JavaScript, Java/Spring, Ruby/Rails, Python, etc.). Maybe I set one up for React development, a different one for Angular development, and a third one for Java/Spring.  Maybe I have one for Cloud/Lambdas and a different one for on-Prem legacy tech. Each "launch" could be configured to include whatever is needed for a particular type of project development. Extensions that are not necessary don't get loaded, thereby saving system resources.

Additionally, these configs could "extend" a base config that includes things you use regardless of stack (e.g. GitLens, themes, icon packs, etc.). Finally, the ability to run the different instances simultaneously if needed would be an added benefit, especially for those that work across multiple teams and projects.

For people that work across multiple stacks, the number of extensions loaded can be daunting, difficult to maintain, and hard to find what you want. Allowing the configurations to be kept small and concise would help mitigate this challenge and provide a cleaner experience.

**Comment by weinand** [comment]
"launch configurations" mentioned in the feature request are not related to the "launch configurations"  provided by VS Code's debug functionality.

**Comment by RodneyBollinger** [comment]
That is correct. This is a new feature request - terminology used in the request is mine and is not based on existing functionality.
